### PR TITLE
Fix bug where ahm_analysis failed when observation groups were disabled

### PR DESCRIPTION
**Pull request by oyvindeide**
Closes: #603 
Closes: #601
Closes: #498

### CODE REVIEW

File: src/semeio/workflows/ahm_analysis/ahmanalysis.py [hunk 1]
```diff
@@ -164,18 +171,24 @@ def run(
         # loop over keys and calculate the KS matrix,
         # conditioning one parameter at the time.
         field_output = {}
+        updated_combinations = deepcopy(combinations)
         for group_name, obs_group in combinations.items():
             print("Processing:", group_name)
 
             #  Use localization to evaluate change of parameters for each observation
-            with tempfile.TemporaryDirectory():
+            with tempfile.TemporaryDirectory(), open_storage("storage", "w") as storage:
```
Comment (sondreso) [inline src/semeio/workflows/ahm_analysis/ahmanalysis.py:179]: Maybe deserves a comment? Something like this
```suggestion
            # The ordering of the context managers matters, we want to open a new storage in the
            # temporary directory
            with tempfile.TemporaryDirectory(), open_storage("storage", "w") as storage:
```

File: src/semeio/workflows/ahm_analysis/ahmanalysis.py [hunk 2]
```diff
@@ -292,8 +298,14 @@ def run(
 
 def _run_ministep(facade, prior_storage, target_storage, obs_group, data_parameters):
     # pylint: disable=too-many-arguments
+    rng = np.random.default_rng(seed=facade.config.random_seed)
```
Comment (sondreso) [inline src/semeio/workflows/ahm_analysis/ahmanalysis.py:301]: This makes me slightly nervous, what is the reasoning?

File: src/semeio/workflows/ahm_analysis/ahmanalysis.py (hunk 1 above)
Comment (sondreso) [reply src/semeio/workflows/ahm_analysis/ahmanalysis.py:179]: RIP formatting...

File: src/semeio/workflows/ahm_analysis/ahmanalysis.py (hunk 2 above)
Comment (oyvindeide) [reply src/semeio/workflows/ahm_analysis/ahmanalysis.py:301]: Why does that make you nervous? It is just so it is reproducible. It was primarily for my snapshot test, but it is just using the seed the user has (optionally) provided.

Comment (oyvindeide) [reply src/semeio/workflows/ahm_analysis/ahmanalysis.py:305]: Created issue for saving random seed in ert: https://github.com/equinor/ert/issues/7499

**Pull request merged**